### ISSUE TITLE
opts not being recongnized

### ISSUE THREAD

**Issue by chuan2984** [Praczet/little-taskwarrior.nvim#1]
hi there, great little plugin. this was like my first thought after installing snacks.

I have just configured it, below are bugs/suggestions

1. opts not being recongized, i tried using opts, using `config = true`, and doing `require 'little-taskwarrior'.setup(opts)`. none of them worked. I wonder the `get_snack_dashboard` is not loading in the default config.
2. the placement of the log file, i have seen people putting it in `~/.local/share/nvim/plugin_name`, this way i wont have to add it to .ignore
3. I notice your comment, and i see it in the code that theres a refresh function now, but it'd be nice to have it refresh periodically so that tasks added outside neovim can also appear without having to relaunch neovim. (not that important)

```
return {
  'folke/snacks.nvim',
  priority = 1000,
  dependencies = {
    {
      'praczet/little-taskwarrior.nvim',
      opts = {
        dashboard = {
          limit = 6,
          debug = false,
          max_width = 56,
          urgency_threshold = 1,
        },
      },
      config = true,
    },
  },
config = function()
  -- whether i do this with empty param or pass in a local copy of the opts, it just do not pick up the opts
  require('little-taskwarrior').setup()
```

**Comment by Praczet** [comment]
Hi, 
Thank you for the message.

ad2) 
I've pushed a fix to change the path of the log file.

For points: 1,3 I need to test it. I'll be back :-D

**Comment by Praczet** [comment]
ad1)

Ok here is my config of snacks.nvim (with Little-taskwarrior in dependencies)

```lua
local myFunc = require("my-functions")

local birth = require("next-birthday")
birth.setup({ bd_file = "/home/adam/Notes/me-social.md" })

local pplBirth = birth.birthdays("now")

local notes = myFunc.SnacksGetRecentNotes(" ", "Recent Notes", 2, 1, 1)
local currentDate = os.date("%Y-%m-%d")

return {
  {
    "folke/snacks.nvim",
    dependencies = {
      {
        "praczet/little-taskwarrior.nvim",
        opts = {
          dashboard = {
            limit = 6,
            debug = false,
            max_width = 56,
            urgency_threshold = 1,
          },
        },
        config = true,
      },
    },
    opts = {
      dashboard = {
        preset = {
          header = myFunc.DashHeader(),
        },
        sections = {
          {
            text = myFunc.DashHeader(),
            align = "center",
            height = 6,
            padding = 1,
          },
          {
            text = { { currentDate, hl = "key" }, { "\n\n" .. table.concat(pplBirth, "\n") .. "\n" } },
            align = "center",
          },
          {
            icon = "",
            title = "Little TaskWarrior",
            pane = 2,
          },
          {
            function()
              local ltw = require("little-taskwarrior")
              return { text = ltw.get_snacks_dashboard_tasks(56, "dir", "special") }
            end,
            pane = 2,
            indent = 2,
          },
          { section = "keys", gap = 1, padding = 1 },
          notes,
          { pane = 2, icon = " ", title = "Recent Files", section = "recent_files", indent = 2, padding = { 1, 1 } },
          { pane = 2, icon = " ", title = "Projects", section = "projects", indent = 2, padding = 1 },
          { pane = 2, section = "startup" },
        },
      },
    },
  },
}
```

This works. I had 6 lines of Tasks. 

My default config look like this (I do not add little-taskwarrior in dependencies (I keep it in other file)). And it works also.

![image](https://github.com/user-attachments/assets/fb7f2438-9d0e-4f3a-981d-d0f28605e5f9)

Can you share your little-taskwarror's config and config for snacks.nvim (Dashboard)?

**Comment by chuan2984** [comment]
@Praczet thanks for the super quick response. heres the [config](https://github.com/chuan2984/dotfiles/blob/dd5adc16637e69739bad438b079ee89c5efca256/.config/nvim/lua/plugins/snacks.lua#L1-L1)
![image](https://github.com/user-attachments/assets/6693bf16-6152-4247-9adc-a882dcd79663)
![image](https://github.com/user-attachments/assets/50064449-e9c6-420d-be99-0f46f7aceb2a)

heres what my dashboard looks like
![image](https://github.com/user-attachments/assets/d3687cc0-ae7d-4dd8-8a1b-c9dc790c0e61)

**Comment by Praczet** [comment]
ok.
Firstly when I changed my config in similar way to you, my config was picked up, but before we start checking this again look at this:

![image](https://github.com/user-attachments/assets/473eefa7-812f-472f-8ee5-ce118d7fc763)

You have to wrap `text` with function in snacks.nvim dashboard configuration.

**Comment by chuan2984** [comment]
thanks! i did exactly what you told me. The interesting now is that the limit works! but everything else does not. i deleted the log file, it still gets populated. the urgency setting was updated a few times, but i dont see the highlight change. Am i still missing something? 

going to sleep now, will respond tomorrow. cheers!

**Comment by Praczet** [comment]
Hi,
Finally I had some time to check it.
And there is what we've missed:

the proper configuration should be like this:

```lua
{
    "folke/snacks.nvim",
    dependencies = {
      {
        "praczet/little-taskwarrior.nvim",
        opts = {
          urgency_threshold = 1,
          dashboard = {
            limit = 6,
            debug = false,
            max_width = 56,
   
          },
        },
        config = true,
      },
    },
// ...
}
```
the `urgency_threshold` is part of plugin configuration not only for the dashboard. So it has to be on the same level as dashboard.

**Comment by chuan2984** [comment]
awesome you are 100% right, and it does update even iyou do tas complete outside of neovim. Thanks a lot!
